From aa8ba118869b75a2a9cd681b2f0362d9d8f1c7ec Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Tue, 12 Jul 2016 11:34:45 +0800
Subject: [PATCH] pax: strip off the trailing slash of file name

When extracting packaes, the trailing slash of the file name
has no means but may cause some issue on system call lstat.
Remove it.

Upstream-Status: Pending

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 src/pat_rep.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/src/pat_rep.c b/src/pat_rep.c
index b9a4636..4cbf6bf 100644
--- a/src/pat_rep.c
+++ b/src/pat_rep.c
@@ -605,7 +605,7 @@ int
 mod_name(ARCHD *arcn)
 {
 	int res = 0;
-
+	char *pt;
 	/*
 	 * Strip off leading '/' if appropriate.
 	 * Currently, this option is only set for the tar format.
@@ -639,6 +639,15 @@ mod_name(ARCHD *arcn)
 	}
 
 	/*
+	 * strip off trailing slash.
+	 */
+	pt = &(arcn->name[strlen(arcn->name) - 1]);
+	if (*pt == '/') {
+		 *pt = '\0';
+		arcn->nlen = strlen(arcn->name);
+	}
+
+	/*
 	 * IMPORTANT: We have a problem. what do we do with symlinks?
 	 * Modifying a hard link name makes sense, as we know the file it
 	 * points at should have been seen already in the archive (and if it
-- 
1.8.5.2.233.g932f7e4

